src/ctap/mod.rs

```diff
@@ -167,18 +189,26 @@ where
             check_user_presence,
             persistent_store,
             pin_protocol_v1,
-            accepts_reset: true,
             #[cfg(feature = "with_ctap1")]
             u2f_up_state: U2fUserPresenceState::new(
                 U2F_UP_PROMPT_TIMEOUT,
                 Duration::from_ms(TOUCH_TIMEOUT_MS),
             ),
+            stateful_command_permission: TimedPermission::granted(now, RESET_TIMEOUT_DURATION),
+            stateful_command_type: Some(StatefulCommand::Reset),
         }
     }
 
-    pub fn check_disable_reset(&mut self, timestamp: Timestamp<isize>) {
-        if timestamp - Timestamp::<isize>::from_ms(0) > Duration::from_ms(RESET_TIMEOUT_MS) {
-            self.accepts_reset = false;
+    pub fn update_command_permission(&mut self, now: ClockValue) {
+        self.stateful_command_permission = self.stateful_command_permission.check_expiration(now);
+    }
+
+    fn check_command_permission(&mut self, now: ClockValue) -> Result<(), Ctap2StatusCode> {
+        self.stateful_command_permission = self.stateful_command_permission.check_expiration(now);
```

Comment:
why not calling `self.update_command_permission(now);` here?

Reply:
Done